src/azul/deployment.py

```diff
@@ -285,13 +284,11 @@ class Terraform:
 
     @cached_property
     def taggable_resources(self) -> Sequence[str]:
-        schema = subprocess.run(['terraform', 'providers', 'schema', '-json'],
-                                cwd=os.path.join(os.environ['project_root'], 'terraform'),
-                                check=True,
-                                capture_output=True,
-                                text=True,
-                                shell=False)
-        schema = json.loads(schema.stdout)
+        schema_path = os.path.join(os.environ['project_root'],
+                                   'terraform',
+                                   '_schema.json')
+        with open(schema_path) as f:
```

Comment:
Consider extracting reading and writing the file into functions.

Reply:
After implementing the other changes you suggested in your review, I don't think this extraction is necessary, so I'm proceeding without it. I'm happy to revisit it if you disagree

Reply:
Hannes ok'd this during standup.